src/NosCore.Controllers/Quest/QTPacketHandler.cs

```diff
@@ -1,45 +1,46 @@
-﻿using System.Threading.Tasks;
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
 using NosCore.GameObject;
 using NosCore.GameObject.Networking.ClientSession;
 using NosCore.GameObject.Providers.QuestProvider;
 using NosCore.Packets.ClientPackets.Quest;
+using NosCore.Packets.Enumerations;
 
 namespace NosCore.PacketHandlers.Quest
 {
     public class QtPacketHandler : PacketHandler<QtPacket>, IWorldPacketHandler
     {
-        private IQuestProvider _questProvider;
+        private readonly IQuestProvider _questProvider;
 
         public QtPacketHandler(IQuestProvider questProvider)
         {
             _questProvider = questProvider;
         }
 
-        public override Task ExecuteAsync(QtPacket qtPacket, ClientSession session)
+        public override async Task ExecuteAsync(QtPacket qtPacket, ClientSession session)
         {
-            //switch (qtPacket.Type)
-            //{
-            //    // On Target Dest
-            //    case 1:
-            //        Session.Character.IncrementQuests(QuestType.GoTo, Session.CurrentMapInstance.Map.MapId, Session.Character.PositionX, Session.Character.PositionY);
-            //        break;
+            var charQuest = session.Character.Quests.FirstOrDefault(q => q.Value.QuestId == qtPacket.Data);
+            if (charQuest.Equals(new KeyValuePair<Guid, CharacterQuest>()))
+            {
+                return;
+            }
 
-            //    // Give Up Quest
-            //    case 3:
-            //        CharacterQuest charQuest = Session.Character.Quests?.FirstOrDefault(q => q.QuestNumber == qtPacket.Data);
-            //        if (charQuest == null || charQuest.IsMainQuest)
-            //        {
-            //            return;
-            //        }
+            switch (qtPacket.Type)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add a 'default' clause to this 'switch' statement.](https://app.codacy.com/manual/NosCoreIO/NosCore/pullRequest?prid=5304333)